src/io/obf.rs

```diff
@@ -60,32 +57,39 @@ where
     let packed_output_size = public_data.packed_output_size;
     let s_bars = sampler_uniform.sample_uniform(&params, 1, d, DistType::BitDist).get_row(0);
     log_mem("Sampled s_bars");
-    let bgg_encode_sampler = BGGEncodingSampler::new(
-        params.as_ref(),
-        &s_bars,
-        sampler_uniform.clone(),
-        obf_params.encoding_sigma,
-    );
-
-    let s_init = &bgg_encode_sampler.secret_vec;
     let t_bar_matrix = sampler_uniform.sample_uniform(&params, 1, 1, DistType::FinRingDist);
     log_mem("Sampled t_bar_matrix");
-
     let hardcoded_key_matrix = M::from_poly_vec_row(&params, vec![hardcoded_key.clone()]);
     log_mem("Sampled hardcoded_key_matrix");
 
     let a = public_data.a_rlwe_bar;
 
     let b = rlwe_encrypt(
         params.as_ref(),
-        sampler_uniform.as_ref(),
+        &sampler_uniform,
         &t_bar_matrix,
         &a,
         &hardcoded_key_matrix,
         obf_params.hardcoded_key_sigma,
     );
 
     log_mem("Generated RLWE ciphertext {a, b}");
+    let m_b = (2 * (d + 1)) * (2 + log_base_q);
```

Comment:
Can we move this and the `error` variables back insie the `p_init` scope? It feels a big vague this `error` variable in the main scope when it is not used anywhere else

Reply:
Ive moved cus to not consume uniform sampler